examples/CMakeLists.txt

```diff
@@ -1,21 +1,26 @@
-# $Id: CMakeLists.txt 100707 2016-10-31 20:21:07Z ihrivnac $
-
 #----------------------------------------------------------------------------
 # Setup the project
 #
-cmake_minimum_required(VERSION 3.3 FATAL_ERROR)
-project(PTL_Examples LANGUAGES C CXX VERSION ${PTL_VERSION})
+cmake_minimum_required(VERSION 3.8...3.20 FATAL_ERROR)
+if(${CMAKE_VERSION} VERSION_LESS 3.12)
+    cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
+endif()
+project(PTL_Examples)
 
-if(("${CMAKE_PROJECT_NAME}" STREQUAL "PTL" OR PTL_BUILD_EXAMPLES) AND
-   CMAKE_CXX_COMPILER_ID MATCHES "(GNU|Clang|AppleClang)")
-    set(CMAKE_CXX_FLAGS "-W -Wall ${CMAKE_CXX_FLAGS}")
+if(CMAKE_CXX_COMPILER_ID MATCHES "(GNU|Clang|AppleClang)")
+    string(APPEND CMAKE_CXX_FLAGS " -W -Wall")
 endif()
 
-# add the common "library"
+find_package(PTL REQUIRED)
+
+# add the common library
+# Gperf and ittnotify options?
 add_subdirectory(common)
-list(APPEND EXTERNAL_LIBRARIES common)
```

Comment:
I think this line is why the CI is failing

Reply:
I've fixed this in the CI script - the examples are now a single project for clarity as there are only a few, so have repointed to the examples/ base directory. Examples that can't be built due to needing externals are automatically disabled if those dependencies aren't found (rotation needs OpenCV for example)